login_server/src/cpp/SingletonManager/PendingTasksManager.cpp

```diff
@@ -192,7 +192,8 @@ void PendingTasksManager::checkForFinishedTasks(Poco::Timer& timer)
 				auto json = transaction->getModel()->getResultJson();
 				bool removeIt = false;
 				if (!json.isNull()) {
-					if (json->get("state").toString() == "success") {
+					auto state = json->get("state");
+					if (!state.isEmpty() && json->get("state").toString() == "success") {
```

Comment:
```suggestion
					if (!state.isEmpty() && state.toString() == "success") {
```

Why do you convert the json to the state variable just to repeat that?

Reply:
you are right